pyo3-build-config/src/impl_.rs

```diff
@@ -604,15 +627,43 @@ impl FromStr for PythonImplementation {
     }
 }
 
-fn is_abi3() -> bool {
+/// Checks if the `abi3` feature is enabled for the PyO3 crate.
+///
+/// Must be called from a PyO3 crate build script.
+pub fn is_abi3() -> bool {
     cargo_env_var("CARGO_FEATURE_ABI3").is_some()
 }
 
+/// Gets the minimum supported Python version from PyO3 `abi3-py*` features.
+///
+/// Must be called from a PyO3 crate build script.
+pub fn get_abi3_version() -> Option<PythonVersion> {
+    let minor_version = (MINIMUM_SUPPORTED_VERSION.minor..=ABI3_MAX_MINOR)
+        .find(|i| cargo_env_var(&format!("CARGO_FEATURE_ABI3_PY3{}", i)).is_some());
+    minor_version.map(|minor| PythonVersion { major: 3, minor })
+}
+
+/// Checks if the `extension-module` feature is enabled for the PyO3 crate.
+///
+/// Must be called from a PyO3 crate build script.
+pub fn is_extension_module() -> bool {
+    cargo_env_var("CARGO_FEATURE_EXTENSION_MODULE").is_some()
+}
+
+/// Checks if we need to link to `libpython` for the current build target.
+///
+/// Must be called from a crate PyO3 build script.
+pub fn is_linking_libpython() -> bool {
+    let target = TargetInfo::from_cargo_env().expect("must be called from a build script");
+
+    target.os == "windows" || target.os == "android" || !is_extension_module()
+}
+
 #[derive(Debug, PartialEq)]
 struct TargetInfo {
```

Comment:
This is becoming much more like [target_lexicon::Triple](https://docs.rs/target-lexicon/latest/target_lexicon/struct.Triple.html), can we just wrap it instead?

Reply:
This is doable, if everyone is OK with adding a new dependency to every build script.
`pyo3-build-config` already re-implements parts of `anyhow` crate to reduce the dependency footprint (I assume).